### PR TITLE
fix(DatePicker): Change the min-width according to the font-size

### PR DESCRIPTION
**Pull request by sivanharel**
### 🔦 Summary

**The issue:** Bookings complained that when reducing the browser window size the circle of the selected day and of the hover state are squashed.

Screen Shot 2021-03-25 at 9 15 10

**What did I change in this PR:**
After investigating we found that the min-width that was defined to 280px (for mobile) is not good and that the min-width should be dependent on the font-size.

In order to align the weekday texts and the day texts, we used the same font definitions for both of them.

Since the min-width on mobile should be 280px, Michal decided to keep this minimum size in the case of font 14px. 
The calculation for the min-width is 280em/14.

We also defined that the supported font sizes are from 14px to 24px. (So in the case of 14px the min-width would be 280px and in the case of 24px the min-width would be 480px).

Screen Shot 2021-03-25 at 9 17 33
Screen Shot 2021-03-25 at 9 17 43

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

File: packages/wix-ui-tpa/src/components/DatePicker/DatePicker.st.css [hunk 1]
```diff
@@ -23,15 +23,14 @@
 :vars {
     calendarWidth: 100%;
     calendarHeight: 100%;
-    sidePadding: 20px;
+    sidePadding: 21px;
```
Comment (sivanharel) [inline packages/wix-ui-tpa/src/components/DatePicker/DatePicker.st.css:26]: It was updated in the spec as well :)

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-ui-tpa**

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_6948/

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-style-react**

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6948

File: packages/wix-ui-tpa/src/components/DatePicker/DatePicker.st.css [hunk 2]
```diff
@@ -214,8 +208,12 @@
 
 .calendar::weekday {
     padding: 20px 0 16px;
-    font: value(CurrentWeekdayFont);
+    font: value(CurrentDayFont);
```
Comment (sivanharel) [inline packages/wix-ui-tpa/src/components/DatePicker/DatePicker.st.css:211]: We decided to define the weekday font according to the `CurrentDayFont`.
This was also updated in the spec.

File: packages/wix-ui-tpa/src/components/DatePicker/DatePicker.st.css [hunk 3]
```diff
@@ -23,15 +23,14 @@
 :vars {
     calendarWidth: 100%;
     calendarHeight: 100%;
-    sidePadding: 20px;
+    sidePadding: 21px;
```
Comment (nitayneeman) [inline packages/wix-ui-tpa/src/components/DatePicker/DatePicker.st.css:27]: Maybe we want to set `14` as a variable?
```suggestion
    minFontSize: 14;
```

I see we use it in 3 calculations (lines 151, 213, 241)

**Review by nitayneeman — APPROVED** [review]
Looks good!

Maybe we want to add one visual or two?
Basically we can screenshot with two different container widths (like we simulate in storybook) - for example 600px & 300px

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

A deploy preview of your PR was successfully released 🖼.

Project Name: **wix-ui-tpa**

You can find it under one of the following URLs:
* https://bo.wix.com/pages/wix-ui-tpa.pr_6948/

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

A deploy preview of your PR was successfully released 🖼.

Project Name: **wix-style-react**

You can find it under one of the following URLs:
* https://preview.wix-style-react.com/pr_6948

**Comment by sivanharel** [comment]
> Maybe we want to add one visual or two?
> Basically we can screenshot with two different container widths (like we simulate in storybook) - for example 600px & 300px

These tests already exist :)

**Pull request merged**